DotNet/README.md

```diff
@@ -0,0 +1,31 @@
+# .NET Style Guide
+
+## Resources
+
+- [CoreFX C# Coding Style](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md)
+- [Framework Design Guidelines](https://msdn.microsoft.com/en-us/library/ms229042(v=vs.110).aspx)
+- [Microsoft C# Coding Conventions](https://msdn.microsoft.com/library/ff926074.aspx)
+
+## Table of Contents
+
+1. [Some title one](#some-title-one)
+
+## Some title one
+
+### Use var instead of explicit types
+
+Exceptions are allowed. For more information read [this fantastic post from Eric Lippert](https://blogs.msdn.microsoft.com/ericlippert/2011/04/20/uses-and-misuses-of-implicit-typing/).
+
+```csharp
+// Bad
+HttpClient httpClient = new HttpClient();
+
+// Good
+var httpClient = new HttpClient();
+
+// Allowed for using an interface instead of a class
+IUserService userService = new UserService();
+
+// Allowed because the variable is poorly named from legacy code
+Graph g = new Graph();

```

Comment:
> BTW, I would like to see an example of this Eric Lippert's sentence:

I have one example in mind where explicit type is _slightly_ more readable than var. (For the record, I prefer `var` 99.99% of the time)

A method receiving a `Func<T1,T2,TResult>`

```
public void SomeMethod(Func<int, int, bool> intComparer)
{
    // do something with intComparer
}
```

```
// Explicit Type
Func<int, int, bool> greaterThanComparer = (x, y) => x > y;
SomeMethod(greaterThanComparer);

// Implicit Type 1 (this won't compile as type cannot be inferred)
var lessThanComparer = (x, y) => x < y;
SomeMethod(lessThanComparer);

// Implicit Type 2 (I find this harder to read than the explicit type version)
var lessThanComparer = new Func<int, int, bool>((x, y) => x < y);
SomeMethod(lessThanComparer);

// Local Function (unusual syntax)
bool equalityComparer(int x, int y) => x == y;
SomeMethod(equalityComparer);
```

Of course this is a rare scenario, and it's OK if someone still prefers the implicit type version..